src/Raven.Server/Documents/Queries/StreamDocumentIndexEntriesQueryResult.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Raven.Client.Documents.Operations.Counters;
+using Raven.Server.Documents.Handlers;
+using Raven.Server.Documents.Includes;
+using Raven.Server.Documents.Queries.Explanation;
+using Raven.Server.ServerWide;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Queries
+{
+    public class StreamDocumentIndexEntriesQueryResult : QueryResultServerSide<BlittableJsonReaderObject>
```

Comment:
It appears this class is the same as `StreamDocumentQueryResult`. Can we exact most of the code to a base class or have a generic `StreamDocumentQueryResult<T>` ?

Reply:
Seems like only `void AddResult` needs to be overridden in both implementations

Reply:
done